src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -233,6 +233,70 @@ public void markDirty()
         colony.markCitizensDirty();
     }
 
+    @Override
+    public boolean equals(final Object o)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1142)